torchgeo/datasets/geo.py

```diff
@@ -418,6 +420,60 @@ def __init__(
         self._crs = cast(CRS, crs)
         self._res = cast(float, res)
 
+    def list_files(self, filename_glob: Optional[str] = None) -> list[str]:
+        """Get list of files matching filename_glob.
+
+        Args:
+            filename_glob: Defaults to self.filename_glob
```

Comment:
Need to add:
```rst
.. versionadded:: 0.5
```
I think we can actually move this method to `GeoDataset` and reuse it in `VectorDataset` instead of duplicating it.

Reply:
If i'm adding support for VectorDataset in a separate PR I will save it for that PR.